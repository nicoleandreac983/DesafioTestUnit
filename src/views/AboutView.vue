<template>
  <div>
    <h1>Soy un ABOUT!</h1>
  </div>
</template>

<style>
@media (min-width: 1024px) {
  .about {
    min-height: 100vh;
    display: flex;
    align-items: center;
  }
}
</style>
