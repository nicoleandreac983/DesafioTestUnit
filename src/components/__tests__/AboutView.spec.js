import {describe, test, expect, it} from "vitest";
import {mount} from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import AboutView from "@/views/AboutView.vue";
import { shallowMount } from '@vue/test-utils'
describe("AboutView", () => {
    test("Probando la existencia del componente o vista AboutView", async () => {
      const router = createRouter({
        history: createWebHistory(),
        routes: [
          {
            path: "/about",
            name: "about",
            component: AboutView,
          },
        ],
      });
   
      router.push("/about");
      await router.isReady();
   
      const wrapper = mount(AboutView, {
        global: {
          plugins: [router],
        },
      });
   
      expect(wrapper.findComponent(AboutView).exists()).toBe(true);
    });
  });


//crear el snapshot
describe('Componente AboutView.vue', () => {
    test('Validacion de match con el snapshot', () => {
      const wrapper = shallowMount(AboutView)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
// snapshot de la estructura HTML del componente AboutView.vue
describe('AboutView', () => {
    const wrapper = shallowMount(AboutView);
    it('should have the correct title', () => {
        expect(wrapper.find('h1').text()).toBe('Soy un ABOUT!');
    });
});
