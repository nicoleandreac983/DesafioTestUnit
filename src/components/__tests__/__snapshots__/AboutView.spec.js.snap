// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente AboutView.vue > Validacion de match con el snapshot 1`] = `
"<div>
  <h1>Soy un ABOUT!</h1>
</div>"
`;
